homeassistant/components/weatherflow/sensor.py

```diff
@@ -0,0 +1,398 @@
+    UnitOfIrradiance,
+    UnitOfLength,
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+    UnitOfVolumetricFlux,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util.unit_system import METRIC_SYSTEM
+
+from .const import DOMAIN, LOGGER
+
+CONCENTRATION_KILOGRAMS_PER_CUBIC_METER = "kg/m³"
+CONCENTRATION_POUNDS_PER_CUBIC_FOOT = "lbs/ft³"
+
+IMPERIAL_UNIT_MAP = {
+    CONCENTRATION_KILOGRAMS_PER_CUBIC_METER: CONCENTRATION_POUNDS_PER_CUBIC_FOOT,
+    UnitOfLength.KILOMETERS: UnitOfLength.MILES,
+    UnitOfPrecipitationDepth.MILLIMETERS: UnitOfPrecipitationDepth.INCHES,
+    UnitOfVolumetricFlux.MILLIMETERS_PER_HOUR: UnitOfVolumetricFlux.INCHES_PER_HOUR,
+    UnitOfPressure.MBAR: UnitOfPressure.INHG,
+    UnitOfSpeed.KILOMETERS_PER_HOUR: UnitOfSpeed.MILES_PER_HOUR,
+}
+
+
+@dataclass
+class WeatherFlowSensorEntityDescription(SensorEntityDescription):
+    """Describes a WeatherFlow sensor entity description."""
+
+    attr: str | None = None
+    conversion_fn: Callable[[Quantity], Quantity] | None = None
+    decimals: int | None = None
+    event_subscriptions: list[str] = field(default_factory=lambda: [EVENT_OBSERVATION])
+    value_fn: Callable[[Quantity], Quantity] | None = None
+
+
+@dataclass
+class WeatherFlowTemperatureSensorEntityDescription(WeatherFlowSensorEntityDescription):
+    """Describes a WeatherFlow temperature sensor entity description."""
+
+    def __post_init__(self) -> None:
+        """Post initialisation processing."""
+        self.native_unit_of_measurement = UnitOfTemperature.CELSIUS
+        self.device_class = SensorDeviceClass.TEMPERATURE
+        self.state_class = SensorStateClass.MEASUREMENT
+        self.decimals = 1
+
+
+@dataclass
+class WeatherFlowWindSensorEntityDescription(WeatherFlowSensorEntityDescription):
+    """Describes a WeatherFlow wind sensor entity description."""
+
+    def __post_init__(self) -> None:
+        """Post initialisation processing."""
+        self.icon = "mdi:weather-windy"
+        self.native_unit_of_measurement = UnitOfSpeed.KILOMETERS_PER_HOUR
+        self.state_class = SensorStateClass.MEASUREMENT
+        self.conversion_fn = lambda attr: attr.to(UnitOfSpeed.MILES_PER_HOUR)
+        self.decimals = 2
+        self.value_fn = lambda attr: attr.to(UnitOfSpeed.KILOMETERS_PER_HOUR)
+
+
+SENSORS: tuple[WeatherFlowSensorEntityDescription, ...] = (
+    WeatherFlowTemperatureSensorEntityDescription(
+        key="air_temperature",
+        translation_key="temperature",
+        has_entity_name=True,
+    ),
+    WeatherFlowSensorEntityDescription(
+        key="air_density",
+        translation_key="air_density",
+        has_entity_name=True,
```

Comment:
Remove it here and set directly on the sensor class

Reply:
Can you elaborate here? I'm a tad confused as to what you want me to do.